react-web/src/components/ConnectWallet/ConnectWallet.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState, useCallback } from "react";
+import './ConnectWallet.scss';
+import Modal from "components/Modal/Modal";
+import Button from "components/Button/Button";
+
+import { useAppDispatch } from "store/store";
+import { getProfileDetails, logout } from "store/slices/auth.slice";
+
+const wallets: Array<string> = ['lace', 'nami', 'yoroi']
+
+declare global {
+    interface Window {
+        cardano:any;
+    }
+}
+let CardanoNS = window.cardano;
+
+const ConnectWallet = () => {
+    const dispatch = useAppDispatch();
+    const [wallet, setWallet] = useState(null)
+    const [address, setAddress] = useState(null)
+    const [isOpen, setIsOpen] = useState(false)
+
+    const openConnectWalletModal = useCallback(() => setIsOpen(true),[])
+
+    const onCloseModal = useCallback(() => setIsOpen(false),[]) 
+
+    const loadWallet = async (walletName: string) => {
+        try {
+            const enabledWallet = await CardanoNS[walletName].enable();
+            setWallet(enabledWallet)
+            if (enabledWallet) {
+                setAddress(await enabledWallet.getChangeAddress())
+            }
+        } catch (err) {
+            // do nothing
+            console.log(err);
+        }
+    }
+
+    useEffect(() => {
+        if (wallet) {
+            const enabledWallet: any = wallet;
+            if (address !== enabledWallet.getChangeAddress()) {
+                // account has been changed. Force logout the user
+                dispatch(logout());
+            } else {
+                // do nothing
+            }
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
```

Comment:
@bogdan-manole , added this to always check from FE if a change in wallet address is made, and if so, make sure to logout the user from the FE.

Reply:
is this working? `[]` will trigger the effect only once when the component will be mounted

Reply:
My bad. corrected now